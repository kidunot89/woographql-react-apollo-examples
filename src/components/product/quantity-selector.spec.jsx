import React from 'react';
import '@testing-library/react/dont-cleanup-after-each';
import { render, cleanup } from '@testing-library/react';

import QuantitySelector from './quantity-selector';

describe('Quantity input control', () => {
  afterAll(() => cleanup());
  let quantity = 0;
  const onIncrement = jest.fn(() => quantity = quantity + 1);
  const onDecrement = jest.fn(() => quantity = quantity - 1);

  const { baseElement } = render(
    <QuantitySelector
      value={quantity}
      onIncrement={onIncrement}
      onDecrement={onDecrement}
    />,
  );

  it('should render successfully', () => {
    expect(baseElement).toMatchSnapshot();
  });
});