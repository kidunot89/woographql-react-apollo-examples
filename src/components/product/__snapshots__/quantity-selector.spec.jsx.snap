// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quantity input control should render successfully 1`] = `
<body>
  <div>
    <div
      class="inline-flex gap-x-3 items-center"
    >
      <span
        class="font-sans font-bold"
      >
        0
      </span>
      <div
        class="relative inline-block w-20"
      >
        <div
          class="appearance-none block text-center leading-5 m-0 py-2.5 px-0 h-8 outline-transparent outline outline-2 outline-offset-2 border rounded quantity-input"
          readonly=""
          type="number"
          value="0"
        />
        <div
          class="icon-button absolute z-10 left-2.5 top-1/2 -translate-y-1/2"
          role="button"
          tabindex="0"
        >
          <i
            class="fa-solid fa-minus"
          />
        </div>
        <div
          class="icon-button absolute z-10 right-2.5 top-1/2 -translate-y-1/2"
          role="button"
          tabindex="0"
        >
          <i
            class="fa-solid fa-plus"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
